docs/guides/test-capability.md

```diff
@@ -0,0 +1,400 @@
+  });
+
+  it('should return a result when called with ...', async () => {
+    const input = {
+      /* Input object as defined in the profile */
+    };
+    const result = await profile.useCases.UseCaseName.perform(input, {
+      provider,
+    });
+
+    expect(result.isOk()).toBe(true);
+  });
+});
+```
+
+</details>
+
+### Asserting result
+
+Usually we want to assert result comming from our map. To do that we can define results manually in each test or use jest snapshots. If you prefer to not use jest snaphots and don't want to manually write testing cases, you can use Superface CLI to generate tests for you, based on test configuration file `sf-test-config.json`.
+
+:::tip CLI HELP
+[Setting up test configuration file](#set-test-configuration-file) will be described in more detail later.
+
+```shell
+npx superface test --generate
+```
+
+:::
+
+:::info
+To use jest snapshots, you have to omit error timestamps before storing snapshot of it. This is resolved in Superface [testing library](#writing-tests-lib) described later.
+:::
+
+```javascript title="profile.provider.test.js" {20-23}
+const { SuperfaceClient } = require('@superfaceai/one-sdk');
+
+describe('scope/profile-name/provider', () => {
+  let sdk, profile, provider;
+
+  beforeAll(async () => {
+    ...
+  })
+
+  it('should return a result when called with ...', async () => {
+    const input = {
+      // ...
+    }
+    const result = await profile.getUseCase('UseCaseName').perform(
+      input,
+      { provider }
+    );
+
+    expect(result.isOk()).toBe(true);
+    expect(result.unwrap()).toEqual({
+      result: "test"
+    });
+  });
+});
+```
+
+### Recording traffic
+
+If you don't want to hit providers API all the time, you can set up recording with `nock` and use mocked responses in development.
+
+There are four modes of `nock.back` recording support and playback, you can read more about them [here](https://github.com/nock/nock#modes). You can also use `nock.recorder.rec()` and `nock.recorder.play()` and handle recordings yourself (more about `rec` and `play` [here](https://github.com/nock/nock#recording)).
+
+```javascript title="profile.provider.test.js" {1,9-10,18,24}
+const nockBack = require('nock').back;
+
+describe('scope/profile-name/provider', () => {
+  let sdk, profile, provider;
+
+  beforeAll(async () => {
+    // ...
+
+    nockBack.fixtures = '/path/to/fixtures/';
+    nockBack.setMode('record');
+  });
+
+  it('should return a result when called with ...', async () => {
+    const input = {
+      // ...
+    };
+
+    const { nockDone } = await back('your-recording.json');
+
+    const result = await profile
+      .getUseCase('UseCaseName')
+      .perform(input, { provider });
+
+    nockDone();
+
+    expect(result.isOk()).toBe(true);
+    expect(result.unwrap()).toEqual({
+      result: 'test',
+    });
+  });
+});
+```
+
+## Test the provider map with Superface testing library {#writing-tests-lib}
```

Comment:
```suggestion
## Use Superface testing library {#testing-lib}
```

Reply:
Other than that... I am lost. I understand that the testing lib is super convenient and does a recording for me etc. But I would like a quick intro what it does and what it needs.